ancpbids/plugins/plugin_dsloader.py

```diff
@@ -64,12 +64,26 @@ def _handle_metadata_files(self, folder):
             return
         for file in filter(lambda f: f.name.endswith(".json"), folder.files):
             if isinstance(file, Artifact):
-                mdfile = MetadataArtifact()
+                if self.options.lazy_loading:
```

Comment:
We do not need to have specific cases for lazy or not lazy mode. If the options has lazy_loaded=True, then we need to set the contents property as a lambda that can resolve itself when the property is accessed.